### DISCUSSION TITLE
GH Actions stuck in queue (for long time and not running)

### DISCUSSION THREAD

**Discussion by ntampakas** [community/community#55538] [Actions]
### Select Topic Area

Bug

### Body

Hey all!

We 've been using GH Actions with ephemeral runners for quite long time now, without any significant problem.
During the last weeks we came up with an issue that pops up quite randomly.

**Issue**:
Random workflows stuck in the queue, waiting for an ephemeral runner to pick up the job. The point is that even though the workflow status is queued, we cannot view this workflow neither by an API call (filtered status=queued) nor by https://github.com/org/repo/actions?query=is%3Aqueued
Seems like the workflow is completely invisible by GH.
As a result the workflow is queued for a long time and not running.

**Workaround**:
Canceling the workflow and re-run it, fixes this issue.

**Comment by Barcov** [comment]
Sadly this does not always work. I'm working with a stuck action that does not respond to anything once it gets stuck. 
Not all of us have permissions to cancel and re-run workflows.

Here's a way to get things moving again in many cases using a dummy commit:
`git commit --amend --no-edit`
`git push --force-with-lease`

**Reply by Pushpinder140891** [reply]
Thanks Barcov, it works for me

**Reply by bmeyers** [reply]
and for me!

**Reply by jakerodriguez-act** [reply]
This worked for me. Funny enough, before pushing another commit, I found this out of concern that I would have two stuck actions. Will the prior action ever get unstuck and try to run in the future?

**Reply by aylensp** [reply]
This worked for me too, the new action run, but the previous action remains stucked.

**Reply by Sharwari19** [reply]
worked for me too! Thanks mate :)

**Reply by gauravdhiman** [reply]
Worked like a charm. Thanks a lot for sharing this !

**Reply by d3xt3r2909** [reply]
Not working for latest issue (30. Oct 2024)

**Reply by cloco-sudhan** [reply]
not working for me to (30. Oct 2024

**Reply by acsant** [reply]
Not working for me either

**Reply by StrangeAJ** [reply]
Same Not working for me too

**Comment by hashimaziz1** [comment]
This didn't work for me even a second time - I now see two stuck Actions - so I checked the [GitHub Status](https://www.githubstatus.com/incidents/m70hk23gx3nx) page and saw that it showed an [ongoing incident](https://www.githubstatus.com/incidents/m70hk23gx3nx) with Actions. I suspect that even in the above cases this is what was responsible, unfortunately nothing to do but wait for GitHub to resolve the incident.

**Reply by IgnisDa** [reply]
Looks like it is working now.

**Reply by hashimaziz1** [reply]
Working for me now too, although a slow-running integration test was still failing, possibly due to latency issues, so I skipped it temporarily and plan to re-enable it in a few hours to make sure.

**Comment by seibelj** [comment]
This is still broken, they say actions is fixed on the status page but it's still not being picked up by runners

**Reply by IgnisDa** [reply]
It worked for me when I cancelled and ran again.

**Reply by arsene-riaworx** [reply]
Been waiting 5 hours. Cancelled and tried again, still queued for 10 mins. Seems their GitHub status is wrong

**Comment by arsene-riaworx** [comment]
Still not running even after cancelling and trying again

**Comment by hashimaziz1** [comment]
Whatever it is/was it looks like it had something to do with this insane, apocalyptic Microsoft outage: 

https://www.bbc.co.uk/news/live/cnk4jdwp49et?page=4

**Comment by sk-chanch** [comment]
stuck with queue

**Comment by quochuydev** [comment]
I can't even cancel the pending job.

**Reply by qxmips** [reply]
same

**Reply by ronnyfm** [reply]
Did any of you find a workaround? I have been trying dummy commits, the GH CLI, nothing works. I cannot remove the ones queued and cannot start new runs.

**Reply by kaiserpc** [reply]
What worked for me was:

* Create a new branch (with a deferent name) based on the base branch of the PR.
* Push the new branch to the repo and create a new PR. 
* Doing above steps the workflows on the original PR get unstuck and start running again.

**Reply by Justina-03** [reply]
Same here.!! Nothing works. Even 'cancel run' failed....

**Reply by bite-weldon** [reply]
Unfortunately, the GitHub action was already in a queued failure but GitHub status page(https://www.githubstatus.com/) showed that everything was working as normal

**Reply by RaveSplash** [reply]
yeah @bite-weldon , right now i'm also facing the same issue where the actions are queue, and i couldn't cancel any run.
While the githubstatus was showing fine

**Reply by AfiqAqil** [reply]
@bite-weldon I'm also facing this issue, tried running a new action/cancel the action, they are stuck are queue.

**Reply by dilip640** [reply]
I'm also facing same.

**Reply by BeThankxz** [reply]
me too!

**Reply by lessGameMoreStudy** [reply]
I'm also

**Reply by tmanastik** [reply]
same

**Reply by inkweon7269** [reply]
me too!!!!

**Reply by jerryamatya** [reply]
me too

**Reply by zameschua** [reply]
same

**Reply by pigbrainfs** [reply]
same

**Reply by wuhuizuo** [reply]
same

**Reply by ckifer** [reply]
+1, in queue for 20+ minutes

**Reply by radic2510** [reply]
same 😢

**Reply by xrex-stone** [reply]
same

**Reply by Brahmah** [reply]
> Unfortunately, the GitHub action was already in a queued failure but GitHub status page(https://www.githubstatus.com/) showed that everything was working as normal @bite-weldon 

same here atm

**Reply by fanna-repowered** [reply]
same

**Reply by Flower-F** [reply]
sad

**Reply by ViswajithVenu** [reply]
same here too.

**Reply by bite-weldon** [reply]
Hi guys, I found an interesting thing, GitHub action page is always in exception（keep yellow dot not green icron）but individual action jobs can be done slowly
![image](https://github.com/user-attachments/assets/29f2ef76-00b0-4643-822d-cbdfad6bc0f3)

**Reply by ViswajithVenu** [reply]
looks like all the steps are getting completed and github is failing to fetch the status and show it as green

**Reply by akashs3curity** [reply]
me too :(

**Reply by prajilal** [reply]
https://www.githubstatus.com/incidents/9yk1fbk0qjjc

**Reply by ajay-anodelabs** [reply]
Same here. inside job detail its saying completed, but on list its just stuck pending

**Comment by dilip640** [comment]
<img width="1014" alt="image" src="https://github.com/user-attachments/assets/a27efedd-8c12-4094-8aad-fea2f6295f84">
is it just a static page?

**Reply by dilip640** [reply]
<img width="216" alt="image" src="https://github.com/user-attachments/assets/4e567315-170e-42a6-ac78-75dcf5911855">

**Reply by blake-ds** [reply]
Same here :)

**Reply by prahladyeri** [reply]
All systems clearly aren't operational. The jobs are getting stuck in the actions queue.

**Reply by ytk2128** [reply]
<img width="212" alt="image" src="https://github.com/user-attachments/assets/4d25f59d-36c0-4b49-b32e-90b073a7a4c0">

**Reply by pt-vamshi** [reply]
same here

**Reply by kevit-chirag-lodariya** [reply]
same here

**Reply by ShockedPlot7560** [reply]
The job is executed but still appears in the queue.

**Reply by viktorlajos1989** [reply]
![image](https://github.com/user-attachments/assets/72985f6a-7fa8-42a0-be23-911ee2dcb569)

**Reply by CPHgroup** [reply]
all queued here :(

**Reply by osama-sajed** [reply]
Guys it looks like the actions itself is running, but it shows up as queued anyways. You can actually check the jobs themselves to see if they are running.

**Reply by anuarora1990** [reply]
:/

**Reply by CPHgroup** [reply]
status page shows degraded event now

**Comment by wuhuizuo** [comment]
stuck in queue again for all my repos and pull requests.

**Comment by prahladyeri** [comment]
There seems to be an ongoing incident with GH Actions right now even though the [status page](https://www.githubstatus.com/) says it's running fine. 

My job is stuck in the queue since last 5-6 minutes. And I'm not even able to cancel the job.

**Reply by wuhuizuo** [reply]
I found out it happened half an hour ago.

**Reply by rtau-cca** [reply]
The incident linked was in July and marked as resolved already. Maybe, they are creating an update for the incident happening now?

**Reply by prahladyeri** [reply]
@rtau-cca  Yes, I've updated the link. The [current status page](https://www.githubstatus.com/) says there is no incident but clearly there is an issue here.

**Reply by sa-ny** [reply]
Same here - stuck in the queue for the past hour.

**Reply by qazz92** [reply]
same...

**Reply by ChrisoCloete** [reply]
same

**Reply by Danish903** [reply]
Stuck in the queue indefinitely...

**Reply by anshuman-sib** [reply]
+1

**Reply by Mufaddal5253110** [reply]
Same

**Reply by ckifer** [reply]
Some of mine are starting

**Reply by arun-s-aot** [reply]
+1

**Reply by Joseunghyo7742** [reply]
+1

**Reply by MananAgr2003** [reply]
+1

**Reply by thinhle-eh** [reply]
+1

**Reply by rtau-cca** [reply]
When I click on the job which was stuck, which will tell you:
>Waiting for a runner to pick up this job...
Job is waiting for a hosted runner to come online.
Some of them are lucky enough to have a runner assigned, and after waiting for some time, say 10 minutes, and refresh the page, I can see some logs of running job appears.

**Reply by rleungx** [reply]
+1

**Reply by nischalstha9** [reply]
+1

**Reply by paungire** [reply]
+1

**Reply by ckifer** [reply]
> Some of them are lucky enough to have a runner assigned, and after waiting for some time, say 10 minutes, and refresh the page, I can see some logs of running job appears.

I have a few of those but now they're stuck at the end of the workflow and won't mark as complete. Perfect :D

**Reply by Kannan-webkyat** [reply]
+1

**Reply by ShimantaKB-Tunnel** [reply]
I'm not able to cancel the workflow run as well.

**Reply by budistwn15** [reply]
same... how long will this problem be resolved?

**Reply by Samuel8000** [reply]
+1 multiple stuck now and can't cancel the workflow... Guess we'll just have to wait

**Reply by jessadaAtUnit** [reply]
same here for a hour :\

**Reply by theibrarkhan** [reply]
+1

**Reply by kevit-chirag-lodariya** [reply]
+1

**Reply by pt-vamshi** [reply]
+1

**Reply by alexey-berdyugin** [reply]
+1

**Reply by nitish-kulkarni-incontact** [reply]
+1

**Reply by kevit-mayank-harnesha** [reply]
Workflow should be finished. Just status is not getting updated.

**Reply by josephinebienes** [reply]
+ 1

**Reply by Usaldudo** [reply]
+1

**Reply by ASMAE20** [reply]
+1

**Reply by MrLongg71** [reply]
+1, What's happened?

**Reply by trilliwon** [reply]
+1

**Reply by youkchansim** [reply]
+1

**Reply by mehul-syncsignature** [reply]
+1

**Reply by NamHillridge** [reply]
+1

**Comment by Blackman99** [comment]
This year I encountered twice. 
Now I can do nothing but wait. 
Microsoft can surely do better

---

Cancel and re-run won't solve this.

**Reply by JoeAnonimist** [reply]
Wait, Github is owned by Microsoft?

**Reply by Futuza** [reply]
> Wait, Github is owned by Microsoft?  

Yeah they bought them in 2018

**Comment by spy16x** [comment]
+1. Multiple actions are stuck.. 

We have self-hosted runners, even with this we are stuck 😐

**Reply by mojtrifork** [reply]
Experiencing this across our entire GitHub organization - all jobs stuck in queue

**Reply by spy16x** [reply]
<img width="143" alt="image" src="https://github.com/user-attachments/assets/cdc31bd7-476b-40d0-a562-c72af945df71">

It's been more than an hour in some cases for us and status page still says all operational

**Comment by EEvacheNN** [comment]
we're facing the same situation!! stuck in queue up to 10 minutes :-(

**Reply by theibrarkhan** [reply]
+1

**Reply by hbaarya** [reply]
+1

**Comment by stas-sbi** [comment]
same here

**Comment by theibrarkhan** [comment]
we're facing the same situation!! stuck in queue up to 30 minutes

**Comment by jay-hsieh** [comment]
Me too ....

**Comment by isgeoffrey** [comment]
Same issue from Hong Kong

**Comment by mhilmiasyrofi** [comment]
same here, stuck for more than 1 hour 😢

**Comment by sunilhonest** [comment]
we're facing the same situation!! stuck in queue for more than 1 hour.

**Comment by bibek4699** [comment]
same here, stuck for more than 90 mins 😢

**Reply by visaruttt** [reply]
+1

**Comment by stalkerg** [comment]
Can somebody call Microsoft?

**Comment by stas-sbi** [comment]
https://downdetector.com/status/github

![Screenshot from 2024-10-30 16-16-00](https://github.com/user-attachments/assets/0a69062e-57f5-475a-912c-783b1816367d)

**Comment by Samuel8000** [comment]
my first action is currently running 🤞

**Reply by ShockedPlot7560** [reply]
Wait for it to be unqueued lmao

**Reply by Samuel8000** [reply]
action completed successfully but the next ones are still stuck

**Comment by sunilhonest** [comment]
githubstatus.com rocks 😄 

<img width="789" alt="image" src="https://github.com/user-attachments/assets/35934022-cafe-4594-b319-7ccfa9e4b7de">

**Comment by edwarddoh** [comment]
+1

**Reply by mrgray8080** [reply]
+1

**Comment by ducvo-kms** [comment]
+1

**Comment by jclapis** [comment]
Stuck here as well across multiple repos.

**Comment by snaps-grace** [comment]
빨리 해결해주세요

**Comment by azamat-palmakhanbetov-m10** [comment]
+1

**Comment by jaderyang-sms** [comment]
kale

**Comment by anilaydinn** [comment]
+1

**Comment by shivjiachnet** [comment]
+1

**Comment by wjdwjdtn92** [comment]
살려주세요

**Comment by abcdefghiraj** [comment]
+1

**Comment by bibek4699** [comment]
Could someone provide the contact information for Bill Gates or Satya Nadella?

**Reply by howetuft** [reply]
+1

**Comment by Usaldudo** [comment]
+1

**Comment by it-AhmedTaha** [comment]
+1

**Comment by aleksei-verbitski-m10** [comment]
+1

**Comment by rio-ap** [comment]
+1

**Comment by mine2424** [comment]
++i

**Comment by SakharamS** [comment]
+1

**Comment by ShockedPlot7560** [comment]
+1

**Comment by abcdefghiraj** [comment]
One of our repos built and deployed successfully a couple of minutes ago. But others are stuck

**Comment by ratibor78** [comment]
same here, actions working but no actual status tracking

**Comment by reyyzzy** [comment]
+1

**Reply by Himanshu-syncsignature** [reply]
+1 ( stuck for hours )

**Comment by dmitry-ianshin-m10** [comment]
+1

**Comment by tuanna1601** [comment]
same here

**Comment by AmirMehrabi** [comment]
+1

**Comment by Himanshu-syncsignature** [comment]
++1

**Comment by josephinebienes** [comment]
+1

**Comment by wisesth** [comment]
+1

**Comment by bytexenon** [comment]
My actions been running for 1 hour now, and I can't even stop them. I wonder how they f*cked up their servers again

**Comment by wisesth** [comment]
+1

**Comment by Bhavesh-Suvalaka** [comment]
+1

**Comment by faizankhan1256** [comment]
my cancelled actions are running haha..

**Comment by PaulChen79** [comment]
+1

**Comment by cherrycharan** [comment]
+1

**Comment by varun-dehaat** [comment]
+1

**Comment by ThumNet** [comment]
+1

**Comment by mbdelarosa** [comment]
+1

**Comment by affanpg** [comment]
+1

**Comment by edberto** [comment]
Raise. +3

**Comment by ThumNet** [comment]
Now they report degraded
![image](https://github.com/user-attachments/assets/1881c451-c4a7-48cb-92db-c0a7c2d7ed2f)

**Reply by Blackman99** [reply]
Finally. 
After somebody's action has been stucked for over an hour...

**Comment by Joseunghyo7742** [comment]
APT APT APT APT

**Reply by HCG-LimSB** [reply]
Uh, uh huh uh huh

**Comment by mine2424** [comment]
Let's all get off work and go home :)

**Reply by faizankhan1256** [reply]
Guys working from home should go to sleep until their workflow runs.

**Reply by mine2424** [reply]
But Github Copilot is still working now....
We have to work with Copilot for NEXT TASK......

**Reply by faizankhan1256** [reply]
let's take it down as well.

**Reply by pongsathorn-92432** [reply]
Wake me up when its runs

**Reply by mine2424** [reply]
actions started alive now :)
We will go back work....

**Comment by AmirMehrabi** [comment]
Current situation with Github Actions is like how my life is going. It's stuck, no one's taking any responsibility.

**Comment by aditya04848** [comment]
++

**Comment by n01-1** [comment]
+1

**Comment by quorcs** [comment]
https://www.githubstatus.com/
https://www.githubstatus.com/history

**Comment by doniyor2109** [comment]
+1

**Comment by suraj-kd-pro** [comment]
+1

**Comment by iharshr** [comment]
+1

**Comment by DannyTheHippo** [comment]
+1

**Comment by shshinbyhcg** [comment]
+1

**Comment by tungnguyensnk** [comment]
+1

**Comment by minhsieh** [comment]
Same here

**Comment by viktorlajos1989** [comment]
I think I get a cup of coffee before I can do the code reviews
https://www.youtube.com/watch?v=xNjyG8S4_kI

**Comment by haneul-hcg** [comment]
Same here

**Reply by anilrajrimal1** [reply]
+1

**Comment by cpx-anish** [comment]
same here

**Comment by RomartM** [comment]
+1

**Comment by erfank245** [comment]
same here its done running but not finishing

**Comment by hallucinogenizer** [comment]
If you're here on 30 Oct, 2024
It's a partial outage and GitHub is working on it.
[See.](https://www.githubstatus.com/)

<img width="1011" alt="image" src="https://github.com/user-attachments/assets/db793f7a-90d0-47d1-a0ab-746fbef97553">

**Comment by Nehemis1615** [comment]
same here

**Comment by ilhamzacky** [comment]
Same Here

**Comment by MrLongg71** [comment]
+1 
I stuck on all my repos...

**Comment by bfwg** [comment]
Same

**Comment by Violetdusk** [comment]
A

**Comment by rashgaroth** [comment]
do i get an action's charge because of this?

**Reply by varbark** [reply]
+1
Also wonder how will it be charged , the job timer is still increasing

**Comment by d3xt3r2909** [comment]
Seems its not working anywhere :)

**Comment by hanjukim-hanwha** [comment]
when it will be available :(

**Reply by osama-sajed** [reply]
@hanjukim-hanwha Have you checked if what you wrote in the actions has already completed or not? It seems the github actions page is showing incorrect information. The actions have run, but it shows them as queued.

**Comment by huxxhu** [comment]
🧋🥹 tea time

**Comment by aman-ansari1** [comment]
+1

**Comment by andyh-lai** [comment]
+1

**Comment by osama-sajed** [comment]
Guys It seems that the **actions themselves running**, they just appear to be stuck in a queue. You all should check this as well, your actions have probably already done what they needed to do.

**Reply by CPHgroup** [reply]
nope, i checked and mine has not run

**Reply by osama-sajed** [reply]
@CPHgroup have you check the job logs? I can see complete job logs even though it is showing up as queued.

**Reply by ratibor78** [reply]
Yepp, my actions are running, I mentioned this up in the tread, there is just a non-actual status of them, but we use self-hosted runner, maybe that's why

**Reply by osama-sajed** [reply]
I am using normal runners and the actions are definitely working.

**Comment by shashank-saxena** [comment]
+1

**Comment by MtkN1** [comment]
When will my pages be deployed? \_(:3」∠)\_

**Reply by osama-sajed** [reply]
Have you tried checking to see if the page was deployed? It looks like the github actions are running but the github website is now showing it as running.

**Reply by MtkN1** [reply]
Thanks for the reply! I checked my page and it is not deployed. The artifact is uploaded, so it may be partially working ¯\\\_(ツ)\_/¯

**Comment by DFilyushin** [comment]
+

**Comment by developerjhp** [comment]
+1

**Comment by KeithGapusan** [comment]
+1

**Comment by mirindra-rak** [comment]
+1

**Reply by autoolops** [reply]
+1

**Comment by taboowiths** [comment]
깃허브 언능 일어나서 해결해줘요 ㅜ

**Comment by DFilyushin** [comment]
See github statys: https://www.githubstatus.com/

**Comment by developerjhp** [comment]
I think the action is working fine, but only the status isn’t updating.

**Reply by osama-sajed** [reply]
Yea that's kinda the conclusion I've come to as well.

**Comment by sksmsWKd** [comment]
resolved?

**Reply by TirushV** [reply]
Not Working still

**Comment by da-amit-mujawar** [comment]
+1 - stuck job runs. Cannot even cancel! GitHub - PLEASE look into this! Not expected for Enterprise level and that too for hours without any visible action!

**Reply by da-amit-mujawar** [reply]
I see github status still shows degraded. One of stuck job completed but 3 still stuck. In any case, this wasted time in terms of Runner as well as developers and lost productivity - GitHub should give free credits to paid customers.

**Comment by josephinebienes** [comment]
I think some of it is working now - one of my "in progress" status has finished. It then started and finish some of my other queued jobs. Although there's still some in progress even though I can see it as completed

**Comment by Hordunlarmy** [comment]
4 stucked actions, won't bloody cancel even

**Comment by a752994118** [comment]
Looks like one of my workflows is in progress now

**Comment by aman-ansari1** [comment]
working now!

**Reply by anilrajrimal1** [reply]
i don't think

**Reply by Hordunlarmy** [reply]
It actually does.. one of my 4 runs completed.. slow asF but works

**Reply by anilrajrimal1** [reply]
yup, now its working

**Reply by wsaeed-tkxel** [reply]
yup working now for me too
was stuck for an hour on a self hosted runner
Did restart
Tried killing processes but nothing worked
And now suddenly its working again.

**Reply by anilrajrimal1** [reply]
see !!
![image](https://github.com/user-attachments/assets/1fdcb57d-8db0-4db5-9e13-7118ea982287)
what do you think of this
![image](https://github.com/user-attachments/assets/4414959f-6bff-44cd-9ee3-5b42d620fb80)
:laughing: :rofl:

**Comment by RaveSplash** [comment]
Just curious is this case a global thing, or just specific region. Does anyone know?

**Comment by kjxbyz** [comment]
+1

<img width="1230" alt="Screenshot 2024-10-30 at 16 33 11" src="https://github.com/user-attachments/assets/fab7a48b-d18c-45d4-8213-9176e18e3e8c">

**Comment by DimuthuSig** [comment]
Failed jobs get updated as failed after a very long time. Successful job don't get updated at all, they either show as queued or in progress.

**Comment by jm4293** [comment]
Why is "Cancel run" also impossible?

**Comment by Benyaminrmb** [comment]
Me too its just showing **waiting for start**.

**Comment by kasipavankumar** [comment]
Looks like it's working now!

**Reply by anilrajrimal1** [reply]
yup !!

**Comment by n01-1** [comment]
Still not working properly. In a two jobs pipeline, one is done but the other is still stuck.

**Comment by autoSteve** [comment]
Not enough o's in smooth to describe it now. Sorted. (Well, actions worked before, just didn't update status, which can have a knock-on effect.) Seems to update status just fine now.

_"Update - We are continuing to investigate delays to status updates to Actions Workflow Runs, Workflow Job Runs, and Check Steps. Customers may see that their Actions workflows have completed, but the run appears to be waiting for its status to update. We will continue providing updates on the progress towards mitigation.
Oct 30, 2024 - 08:48 UTC"_
